### DISCUSSION TITLE
With transpilePackages, how can I use local tsconfigs ?

### DISCUSSION THREAD

**Discussion by 2wheeh** [vercel/next.js#55034] [Help]
### Summary

I'm setting up a monorepo using `yarn workspace` with the following structure:
```
.
├── .yarn
├── pakcage.json
└── packages/
    ├── my-app (next)/ 
    │   ├── tsconfig.json
    │   ├── next.config.json 
    │   ├── package.json (@mono/my-app)
    │   └── ...
    ├── ui/
    │   ├── tsconfig.json
    │   ├── package.json (@mono/ui)
    │   └── ...
    └── utils/
        ├── tsconfig.json
        ├── package.json (@mono/utils)
        └── ...

```
Both `@mono/ui` and `@mono/utils` contain `.ts` files.
`@mono/my-app` depends on both packages and transpiles them with the following config :

```js
// next.config.json
/** @type {import('next').NextConfig} */
const nextConfig = {
  transpilePackages: ['@mono/utils', '@mono/ui'],
};

module.exports = nextConfig;
```
My question is how to make Next.js use the local `tsconfig` of each package when transpiling.

For example, when `@mono/ui`  has a `paths` field in its `tsconfig`  and use it, `@mono/my-app` fails to transpile since Next.js doesn't know this `tsconfig` from `@mono/ui` and only knows `tsconfig` in `@mono/my-app`.

And when running `next build`, it checks for type errors in all dependencies in transpilePackages, including itself, using just `tsconfig` in `@mono/my-app`. 

I found it happens same on Vercel Solutions' [monorepo example](https://github.com/vercel/examples/tree/9b2e5fbda9f7a972f35e0a2a5675bcedc0e3465b/solutions/monorepo).

How can I configure Next.js to use the tsconfig of each package when using the transpilePackages option, without transpiling each package independently using `tsc`? 

For now, I'm thinking of using `ignoreBuildErrors: true` and typeChecking on IDE.

### Additional information

My question seems to be similar to : 
1. https://github.com/vercel/next.js/discussions/50866
2. https://github.com/vercel/next.js/discussions/33813#discussioncomment-6062173

**Comment by mstade** [comment]
I'm running into this issue as well, did you find any solution?

**Comment by BrentChesny** [comment]
I'm also looking for a solution to this problem.

**Comment by 2wheeh** [comment]
@mstade @BrentChesny Hi there! I've been working on:
1. **not using path aliases in common packages** 
2. utilizing `@typescript-eslint` to type check during development
3. opting in `ignoreBuildErrors: true` to skip type checking during build time

I'm not happy with this workaround 😕. 
Is there any support or feature development planned for this? @timneutkens

**Comment by frontend-dohun-kim** [comment]
Any update on this?

**Reply by frontend-dohun-kim** [reply]
In my opinion, If you choose not to build, we won't be able to using path alias.

**Comment by BarakChamo** [comment]
Would be great to get an answer on this, path aliases work fine during development but fail when trying to build.
